src/virtual_memory/anonymous.cpp

```diff
@@ -57,7 +55,7 @@ namespace argo {
 				throw std::system_error(std::make_error_code(static_cast<std::errc>(errno)), msg_main_mmap_fail);
 				exit(EXIT_FAILURE);
 			}
-			char* virtual_addr = (char*) 0x300000000000l;
+			char* virtual_addr = (char*) ARGO_START + ARGO_SIZE/2l;
```

Comment:
Adjusting the limit between backing storage and ArgoDSM memory dynamically, so that we don't have to change it again when the 8TB limit is fixed later.